docs/versioning.md

```diff
@@ -254,6 +261,195 @@ existing CRD, it will typically be defined in `v1alpha1`. In some rare cases, a
+Skip upgrading CRD "antreacontrollerinfos.crd.antrea.io" since it only has one version.
+Upgrading 6 objects of CRD "tiers.crd.antrea.io".
+Successfully upgraded 6 objects of CRD "tiers.crd.antrea.io".
+
+# You can now upgrade to Antrea v2.0 successfully.
+kubectl apply -f https://github.com/antrea-io/antrea/releases/download/v2.0.0/antrea.yml
+```
+
+#### Case 4: upgrading from Antrea v1.12 (or older) with Helm
+
+```
+# Start by upgrading to Antrea v1.15.1, starting with CRDs.
+kubectl apply -f https://github.com/antrea-io/antrea/releases/download/v1.15.1/antrea-crds.yml
+helm upgrade antrea antrea/antrea --namespace kube-system --version 1.15.1
+
+# Download antctl from the v1.15.1 release assets.
+$ antctl version
+antctlVersion: v1.15.1
+controllerVersion: v1.15.1
+
+# Upgrade API storage for all CRDs.
+# For usage information run: antctl upgrade api-storage --help
+# In particular, the script will upgrade the default Tiers managed by Antrea to
+# the new stored version (v1beta1).
+$ antctl upgrade api-storage
+Skip upgrading CRD "externalnodes.crd.antrea.io" since it only has one version.
+Skip upgrading CRD "trafficcontrols.crd.antrea.io" since it only has one version.
+Skip upgrading CRD "externalentities.crd.antrea.io" since all stored objects are in the storage version.
+Skip upgrading CRD "supportbundlecollections.crd.antrea.io" since it only has one version.
+Skip upgrading CRD "antreaagentinfos.crd.antrea.io" since it only has one version.
+Skip upgrading CRD "ippools.crd.antrea.io" since it only has one version.
+Skip upgrading CRD "antreacontrollerinfos.crd.antrea.io" since it only has one version.
+Upgrading 6 objects of CRD "tiers.crd.antrea.io".
+Successfully upgraded 6 objects of CRD "tiers.crd.antrea.io".
+
+# You can now upgrade to Antrea v2.0 successfully, starting with CRDs.
+kubectl apply -f https://github.com/antrea-io/antrea/releases/download/v2.0.0/antrea-crds.yml
+helm upgrade antrea antrea/antrea --namespace kube-system --version 2.0.0
+```
+
+### Other upgrade considerations
+
+Some deprecated options have been removed from the Antrea configuration:
+
+* `nplPortRange` has been removed from the Agent configuration, use
+  `nodePortLocal.portRange` instead.
+* `enableIPSecTunnel` has been removed from the Agent configuration, use
+  `trafficEncryptionMode` instead.
+* `multicastInterfaces` has been removed from the Agent configuration, use
+  `multicast.multicastInterfaces` instead.
+* `multicluster.enable` has been removed from the Agent configuration, as the
+  Multicluster functionality is no longer gated by a boolean parameter.
+* `legacyCRDMirroring` has been removed from the Controller configuration, as it
+  dates back to the v1 major version bump, and it has been ignored for years.
+
+If you are porting your old Antrea configuration to Antrea v2, please make sure
+that you are no longer using these parameters. Unknown parameters will be
+ignored by Antrea, and the behavior may not be what you expect.
```

Comment:
@tnqn I think that thanks to one of your recent changes, unknown parameters will indeed be ignored instead of crashing the Agent?

Reply:
Yes, I confirmed they are ignored.